### PR TITLE
[server/db] update hackerProfile model to reflect migrations (and added role field) 

### PR DESCRIPTION
**Pull request by WilhelmWillie**
* Updated model definition in `models/hackerProfile.js` to reflect changes to schema from migrations
* Added `role` field to hackerProfile (so we can set up some role-check middleware to prevent hackers from accessing too much data)

### CODE REVIEW

**Comment by WilhelmWillie** [comment]
@saresend was planning on creating an auth middleware to check if the current user is an admin or not, realized the model wasn't updated and added the role field

i had to change  up `.sequelizerc` to make it work on my part though, not sure if we're doing migrations and stuff the same way so lmk if this works for you

File: .sequelizerc [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
 module.exports = {
-  'config': path.resolve('src/config', 'database.js'),
-  'models-path': path.resolve('src/models'),
-  'seeders-path': path.resolve('src/seeders'),
-  'migrations-path': path.resolve('src/migrations')
+  'config': path.resolve('api/database', 'database.js'),
+  'models-path': path.resolve('api/models'),
+  'seeders-path': path.resolve('./seeders'),
+  'migrations-path': path.resolve('./migrations')
```
Comment (WilhelmWillie) [inline .sequelizerc]: might be worth moving `seeders` and `migrations` inside the `api/database` folder

Comment (saresend) [reply .sequelizerc]: Yeah, that makes sense to me

**Comment by saresend** [comment]
That's probably fine, I've generally been running against the mysql url directly rather than from the config, so if the .sequelizerc needed fixing that super makes sense

**Comment by WilhelmWillie** [comment]
> That's probably fine, I've generally been running against the mysql url directly rather than from the config, so if the .sequelizerc needed fixing that super makes sense

gotcha, cool then i'll go ahead and move migrations and seeders inside the `api/database` folder too

**Comment by saresend** [comment]
The privileged hackerProfile access was something I was thinking about recently and I wanted to hear your throughts on a slightly different approach to handling this issue. It would involve basically having an entirely separate table to denote info about hackers that the hackers themselves can't write to. I'm not sure if right now there's huge value add to it since realistically it's just the status, but down the line it might be valuable for things like judging as well, and would make it possible to write middlewares against sql tables, rather than having to stratify access schemes against specific columns in our database.

**Comment by WilhelmWillie** [comment]
> The privileged hackerProfile access was something I was thinking about recently and I wanted to hear your throughts on a slightly different approach to handling this issue. It would involve basically having an entirely separate table to denote info about hackers that the hackers themselves can't write to. I'm not sure if right now there's huge value add to it since realistically it's just the status, but down the line it might be valuable for things like judging as well, and would make it possible to write middlewares against sql tables, rather than having to stratify access schemes against specific columns in our database.

Hmm thought about this, so in parallel to `HackerProfiles`, we'd have a table like `HackerPermissions` (name TBD). This could have columns like `isAdmin`, `isSponsor`, `isJudge` (which IMO might be best because we might have users who are both sponsors + judges, a role enum wouldn't allow this)

For application review, we'll prob have a separate table called `HackerReviews` with a `userId`, `reviewerId`, `reviewScore`, `reviewComments`, etc. 

Thoughts?

**Comment by saresend** [comment]
Actually, let me take a look at how auth0 roles might work for this, so ideally we can just get the user role on the session object and work off of that. Immediately though, sounds like you'd rather leave the user status on the `HackerProfile` table?

**Comment by WilhelmWillie** [comment]
Oh if we can get the role from the session object, that would be money. For now, let's roll with this user status column and adjust based off what auth0 can provide, just so we can spin something up now

**Pull request merged**